libpromises/dbm_lmdb.c

```diff
@@ -79,13 +79,18 @@ static int DB_MAX_READERS = -1;
 /******************************************************************************/
 
 static void HandleLMDBCorruption(MDB_env *env, const char *msg);
+static void HandleFullLMDB(MDB_env *env);
 
-static inline void CheckLMDBCorrupted(int rc, MDB_env *env)
+static inline void CheckLMDBUsable(int rc, MDB_env *env)
```

Comment:
I know you didn't introduce this, but IMO in cases like this a const is very nice, to prevent something like an accidental `if (rc = 1)` (always true):

```suggestion
static inline void CheckLMDBUsable(const int rc, MDB_env *env)
```

Reply:
This is not inline with our code style / guidelines:

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#safety--correctness

I don't agree that it loses the point - it effectively communicates and enforces that this variable doesn't change through the function. I do agree that it's more important for larger chunks of code. I find it very helpful to see a variable and know that it's written to exactly once, and where / what value that is. Also, using language feature to prevent accidents like the one mentioned is good. Having checks is also nice, of course.

> We (should) have checks for this.

Maybe we do, maybe we don't. There are probably other examples, that might be harder to check for. The point is that it's safer and more readable / maintainable, IMO.

Not a hard requirement from my side, especially since you didn't introduce it, but I don't agree with the reasoning.

Reply:
^^

Reply:
Then we don't agree. And you are free to improve my code if you think it needs it.

Reply:
> This is not inline with our code style / guidelines: https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#safety--correctness

IMHO, the guidelines actually refer to the case where you have `const *` parameters (you cannot alter a non-pointer parameter, of course, only the local copy of it).